health.c

```diff
@@ -528,6 +528,7 @@ int aws_health_listener_destroy(pool *p, struct aws_health *health) {
 
   if (health->conn != NULL) {
     pr_inet_close(health->pool, health->conn);
+    health->conn = NULL;
```

Comment:
I encountered an unexpected segfault, running under `gdb`, due to this pointer not be null as expected after we close the connection.